### PR TITLE
fix(rdb): pgaudit

### PR DESCRIPTION
**Pull request by ldecarvalho-doc**

### CODE REVIEW

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+The `pgaudit` extension is available with Scaleway Managed Databases for PostgresSQL. The extension is a powerful monitoring and logging tool that allows you to keep track of the actions happening in your databases, and record them.
+
+
+`pgaudit` allows you to log different statements executed in your Database Instance, or specific databases. You can keep track of who accessed your databases, what actions were performed, and when, under a [structured format](https://github.com/pgaudit/pgaudit/blob/master/README.md#format).
+
+
+You can configure the extension to log only what is relevant to your audits.
+
+
+## Requirements
```
Comment (jcirinosclwy) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:27]:
```suggestion
<Macro id="requirements" />

```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+- Be the [Owner](/identity-and-access-management/iam/concepts/#owner) of the Organization in which the actions will be performed, or an IAM user with the [necessary permissions](/identity-and-access-management/iam/concepts/#permission)
```
Comment (jcirinosclwy) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:28]: - A Scaleway account logged into the [console](https://console.scaleway.com)

Comment (jcirinosclwy) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:28]:
```suggestion
- A Scaleway account logged into the [console](https://console.scaleway.com)
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+- A Scaleway account logged into the [console](https://console.scaleway.com)
```
Comment (jcirinosclwy) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:29]:
```suggestion
- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+   4. Click <Icon name="edit" />.
+   5. Click **+ Add parameters**.
+   6. Select `pgaudit.log` in the drop-down.
+   7. Enter one or more of the statement class values listed above.
+     <Message type="note">
+       The values must be written in uppercase. If entering more than one class, separate them with a comma and no spaces. For example: `READ,WRITE`.
+     </Message>
+   8. Click <Icon name="validate" /> to validate.
+     <Message type="note">
+       The configuration takes a few seconds to be applied. During this time the Database Instance connection remains uninterrupted. However, you must wait the few seconds if you want to edit your advanced settings again.
```
Comment (jcirinosclwy) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:107]:
```suggestion
       The configuration takes a few seconds to be applied. During this time the Database Instance connection remains uninterrupted. However, you must wait a few seconds if you want to edit your advanced settings again.
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+    <Message type="tip">
+      Run `\du` to see a list of your database roles and check that the new role was created.
+    </Message>
+3. Set the role you created as `pgaudit.role`
+    ```
+    SET pgaudit.role = 'auditore';
+    ```
+    <Message type="note">
+      Alternatively you can follow these steps to confiure `pgaudit.role` from the Scaleway console:
```
Comment (jcirinosclwy) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:181]:
```suggestion
      Alternatively, you can follow these steps to configure `pgaudit.role` from the Scaleway console:
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+
+      1. Go to the **Advanced settings** of your Database Instance in the Scaleway console.
+      2. Click <Icon name="edit" />.
+      3. Click **+ Add parameters**.
+      4. Select `pgaudit.role` in the drop-down and type the name of the role you created to be the principal.
+      5. Click <Icon name="validate" /> to validate.
+          <Message type="note">
+            The configuration takes a few seconds to be applied. During this time the Database Instance connection remains uninterrupted. However, you must wait the few seconds if you want to edit your advanced settings again.
```
Comment (jcirinosclwy) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:189]:
```suggestion
            The configuration takes a few seconds to be applied. During this time the Database Instance connection remains uninterrupted. However, you must wait a few seconds if you want to edit your advanced settings again.
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+
+You can set up
+
+
+Object renames are logged under the name they were renamed to. (https://github.com/pgaudit/pgaudit/blob/master/README.md#caveats)
+
+
+### Optimizing disk usage
+
+- Audit logs are flushed and consume disk space. The [log ingestion rate](https://www.scaleway.com/en/docs/observability/cockpit/reference-content/cockpit-limitations/#limits-for-loki) is **4MB per second**.
```
Comment (jcirinosclwy) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:215]:
```suggestion
- Audit logs are flushed and consume disk space. The [log ingestion rate](https://www.scaleway.com/en/docs/observability/cockpit/reference-content/cockpit-limitations/#limits-for-loki) is **4 MB per second**.
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+    You can edit the `pgaudit.role` values in the advanced settings anytime.
+4. Run the following command in your Database Instance to grant the principal audit role permissions:
+    ```
+    GRANT PERMISSION1, PERMISSION2
+      on <object>
+      to <principal-role>;
+    ```
+    Replace `<PERMISSION1>` AND `<PERMISSION2>` with one or more of the supported permissions, `<object>` with the name of the data object you want to audit, and `auditor` with the `<principal-role>` with the name of the principal role.
+
+If the operation was successful you see `GRANT` as an output.
```
Comment (jcirinosclwy) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:201]:
```suggestion
If the operation was successful, you see `GRANT` as an output.
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+
+ <TabsTab label="For a specific database">
+
+
+   1. [Connect to your database](/managed-databases/postgresql-and-mysql/how-to/connect-database-instance/) as an admin user.
+   2. Run the following command to set `pgaudit.log` for the database. Replace `<database>` with the name of your database, and ´<class>` with one or more of the statement class values listed above.
+     ```
+     database=> ALTER DATABASE <database> SET pgaudit.log = '<class>'
+     ```
+     If the operation is successful no output is returned.
```
Comment (jcirinosclwy) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:121]:
```suggestion
     If the operation is successful, no output is returned.
```

**Review by jcirinosclwy — APPROVED** [review]

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 10]
```diff
@@ -0,0 +1,307 @@
...
+9. Run the following command to install the `pgaudit` extension.
+   ```
+   CREATE EXTENSION pgaudit
+   ```
+
+   If the operation is successful, no output is returned.
+
+
+## Configuring pgaudit
+
```
Comment (bene2k1) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:52]:
```suggestion
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 11]
```diff
@@ -0,0 +1,307 @@
...
+
+`pgaudit` provides two different types of logging:
+   - **Session audit logging** - logs the operations, such as queries and updates, executed by a user during a session within a database or the entire Database Instance
+   - **Object audit logging** - logs actions that involve specific data objects, like tables or files.
+
```
Comment (bene2k1) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:57]:
```suggestion
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 12]
```diff
@@ -0,0 +1,307 @@
...
+
+To use each one, you must configure the `pgaudit.log` and `pgaudit.role` settings, respectively.
+
```
Comment (bene2k1) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:60]:
```suggestion
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 13]
```diff
@@ -0,0 +1,307 @@
...
+
+Both settings can be set up at the Database Instance level, meaning that a global configuration can be applied to all databases within the Instance. You can configure them in the [advanced settings](/managed-databases/postgresql-and-mysql/concepts/#advanced-settings) of your Database Instance. You can use both settings concurrently.
+
```
Comment (bene2k1) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:63]:
```suggestion
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 14]
```diff
@@ -0,0 +1,307 @@
...
+
+`pgaudit.log` can also be configured for specific databases within your Instance.
+
```
Comment (bene2k1) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:66]:
```suggestion
```

Is there are specific reason why you used two line breaks every time?

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 15]
```diff
@@ -0,0 +1,303 @@
...
+</Tabs>
+
+### Setting up pgaudit.role
+
+Object audit logging is configured through the roles system. The `pgaudit.role` setting allows you to specify a principal audit role (also known as "master role").
+
+Various audit roles can be established, delegating responsibility for distinct aspects of audit logging to different groups. These roles must subsequently be assigned to the principal audit role to facilitate audit logging.
+
+The primary function of the principal audit role is to oversee and manage the auditing process. The auditing process is generally handled by an independent individual or company, which will use the audit logs to perform their audit.
```
Comment (scw-wtimmermans) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:144]: to oversee and manage the audit logging process*

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 16]
```diff
@@ -0,0 +1,303 @@
...
+    CREATE ROLE auditor;
+    ```
+    You can replace `auditor` with your name of choice.
+
+    <Message type="tip">
+      Run `\du` to see a list of your database roles and check that the new role was created.
+    </Message>
+3. Set the role you created as `pgaudit.role`
+    ```
+    SET pgaudit.role = 'auditore';
```
Comment (scw-wtimmermans) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:176]: auditor*

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx (hunk 15 above)
Comment (ldecarvalho-doc) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:144]:
```suggestion
The primary function of the principal audit role is to oversee and manage the audit logging process. The auditing process is generally handled by an independent individual or company, which will use the audit logs to perform their audit.
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx (hunk 16 above)
Comment (ldecarvalho-doc) [reply managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:176]:
```suggestion
    SET pgaudit.role = 'auditor';
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 17]
```diff
@@ -0,0 +1,301 @@
...
+  - Audit logs are displayed in a specific format that you can also find referenced in the [official pgaudit repository on Github](https://github.com/pgaudit/pgaudit/blob/master/README.md#format)
+
+### Optimizing disk usage
+
+- Audit logs are flushed and consume disk space. The [log ingestion rate](https://www.scaleway.com/en/docs/observability/cockpit/reference-content/cockpit-limitations/#limits-for-loki) is **4 MB per second**.
+
+  To avoid log generation from exceeding ingestion, and the subsequent uncontrolled growth of disk usage, make sure you closely monitor the disk usage of the Database Instance.
+
+  <Message type="important">
+    If the disk space becomes full, some audit logs may be lost.
```
[the remark this answers is not in the document]
Comment (ldecarvalho-doc) [reply managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:231]:
```suggestion
    If the disk space becomes full, some audit logs may be lost, and the database can become unavailable. 
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 18]
```diff
@@ -0,0 +1,303 @@
...
+tags: managed-database postgresql pgaudit pg-extensions
+dates:
+ validation: 2024-03-05
+ posted: 2024-03-05
+categories:
+ - managed-databases
+ - postgresql-and-mysql
+---
+
+The `pgaudit` extension is available with Scaleway Managed Databases for PostgresSQL. The extension is a powerful monitoring and logging tool that allows you to keep track of the actions happening in your databases, and record them.
```
Comment (ldecarvalho-doc) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:17]:
```suggestion
The `pgaudit` extension is available with Scaleway Managed Databases for PostgreSQL. The extension is a powerful monitoring and logging tool that allows you to keep track of the actions happening in your databases, and record them.
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 19]
```diff
@@ -0,0 +1,303 @@
...
+1. Click **PostgreSQL and MySQL** under **Managed Databases** on the side menu of the Scaleway console. A list of your Database Instances displays.
+2. Click the Database Instance name or <Icon name="more" /> > **More info** to access the Instance information page.
+3. Click the **Advanced settings** tab.
+4. Click <Icon name="edit" />.
+5. Click **+ Add parameters**.
+6. Select `rdb.enable_pgaudit` in the drop-down and click the toggle to set it to `on`.
+7. Click <Icon name="validate" /> to validate. Your Database Instance restarts.
+
+   <Message type="important">
+     Restarting can take up to a few dozen seconds depending on the size of your Database Instance.
```
Comment (SamyOubouaziz) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:42]: maybe something generic than "few dozen"

Comment (ldecarvalho-doc) [reply managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:42]: cc @scw-wtimmermans what do you think?

Comment (ldecarvalho-doc) [reply managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:42]: I prefer to be specific, personally

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 20]
```diff
@@ -0,0 +1,303 @@
...
+- A [Database Instance](/managed-databases/postgresql-and-mysql/how-to/create-a-database) running a PostgreSQL engine
+
+## Enabling and installing pgaudit
+
+1. Click **PostgreSQL and MySQL** under **Managed Databases** on the side menu of the Scaleway console. A list of your Database Instances displays.
+2. Click the Database Instance name or <Icon name="more" /> > **More info** to access the Instance information page.
+3. Click the **Advanced settings** tab.
+4. Click <Icon name="edit" />.
+5. Click **+ Add parameters**.
+6. Select `rdb.enable_pgaudit` in the drop-down and click the toggle to set it to `on`.
```
[the remark this answers is not in the document]
Comment (SamyOubouaziz) [reply managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:38]: (I can't see the console for this option)

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 21]
```diff
@@ -0,0 +1,303 @@
...
+
+The possible values include:
+
+| Class        | Action         |
+| ------------ | ------------------- |
+| `READ`       | Logs `SELECT` and `COPY` commands when the source is a relation or a query. |
+| `WRITE`      | Logs `INSERT`, `UPDATE`, `DELETE`, `TRUNCATE`, and `COPY` commands when the destination is a relation. |
+| `FUNCTION`   | Logs function calls and `DO` blocks. |
+| `ROLE`       | Logs statements related to roles and privileges, such as `GRANT`, `REVOKE`, `CREATE`, `ALTER`, and `DROP` `ROLE`.        |
+| `DDL`        | Covers all Data Definition Language (DDL) statements not included in the ROLE class. |
```
[the remark this answers is not in the document]
Comment (ldecarvalho-doc) [reply managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:79]:
```suggestion
| `DDL`        | Covers all Data Definition Language (DDL) statements not included in the `ROLE` class. |
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 22]
```diff
@@ -0,0 +1,303 @@
...
+   2. Click <Icon name="edit" />.
+   3. Click **+ Add parameters**.
+   4. Select `pgaudit.log` in the drop-down.
+   5. Enter one or more of the statement class values listed above.
+      <Message type="note">
+        The values must be written in uppercase. If entering more than one class, separate them with a comma and no spaces. For example: `READ,WRITE`.
+      </Message>
+   6. Click <Icon name="validate" /> to validate.
+      <Message type="note">
+        The configuration takes a few seconds to be applied. During this time the Database Instance connection remains uninterrupted. However, you must wait the few seconds if you want to edit your advanced settings again.
```
Comment (SamyOubouaziz) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:101]: "wait the few seconds" seems a bit off, maybe

```suggestion
        The configuration takes a few seconds to be applied. During this time the Database Instance connection remains uninterrupted. However, you must wait until the new configuration is applied to edit your advanced settings again.
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx (hunk 19 above)
Comment (ldecarvalho-doc) [reply managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:42]:
```suggestion
     Restarting can take several seconds depending on the size of your Database Instance.
```

what about this?

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 23]
```diff
@@ -0,0 +1,303 @@
...
+    <Message type="note">
+      Alternatively, you can follow these steps to configure `pgaudit.role` from the Scaleway console:
+
+      1. Go to the **Advanced settings** of your Database Instance in the Scaleway console.
+      2. Click <Icon name="edit" />.
+      3. Click **+ Add parameters**.
+      4. Select `pgaudit.role` in the drop-down and type the name of the role you created to be the principal.
+      5. Click <Icon name="validate" /> to validate.
+        <Message type="note">
+          The configuration takes a few seconds to be applied. During this time the Database Instance connection remains uninterrupted. However, you must wait the few seconds if you want to edit your advanced settings again.
```
Comment (ldecarvalho-doc) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:187]:
```suggestion
The configuration takes a few seconds to be applied. During this time the Database Instance connection remains uninterrupted. However, you must wait until the new configuration is applied to edit your advanced settings again.
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 24]
```diff
@@ -0,0 +1,303 @@
...
+      </Message>
+   You can edit the `pgaudit.log` values in the advanced settings anytime.
+ </TabsTab>
+
+
+ <TabsTab label="For a specific database">
+
+
+   1. [Connect to your database](/managed-databases/postgresql-and-mysql/how-to/connect-database-instance/) as an admin user.
+   2. Run the following command to set `pgaudit.log` for the database. Replace `<database>` with the name of your database, and `<class>` with one or more of the statement class values listed above.
```
[the remark this answers is not in the document]
Comment (ldecarvalho-doc) [reply managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:111]: oops forgot
```suggestion
   2. Run the following command to set `pgaudit.log` for the database. Replace `<database>` with the name of your database, and `<class>` with one or more of the statement class values listed above. When listing more than one class, separate them with a comma and no spaces. For example: `WRITE,FUNCTION`.
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 25]
```diff
@@ -0,0 +1,303 @@
...
+    ```
+    SET pgaudit.role = 'auditor';
+    ```
+    <Message type="note">
+      Alternatively, you can follow these steps to configure `pgaudit.role` from the Scaleway console:
+
+      1. Go to the **Advanced settings** of your Database Instance in the Scaleway console.
+      2. Click <Icon name="edit" />.
+      3. Click **+ Add parameters**.
+      4. Select `pgaudit.role` in the drop-down and type the name of the role you created to be the principal.
```
Comment (SamyOubouaziz) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:184]:
```suggestion
      4. Select `pgaudit.role` in the drop-down and enter the name of the role you created to be the principal.
```

Comment (SamyOubouaziz) [reply managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:184]: for consistency

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 26]
```diff
@@ -0,0 +1,303 @@
...
+    ```
+    Replace `mydb` with the name of your database.
+
+### Session audit logging and object audit logging
+
+To set up:
+  - session audit logging of all statements except `READ` and `WRITE` for all databases and roles, and
+  - object audit logging of `SELECT`, `INSERT`, `UPDATE`, and `DELETE` statements only on a specific table in a database.
+
+1. Set `pgaudit.log` to `FUNCTION,ROLE,DDL,MISC,MISC_SET` and the `pgaudit.role` as `auditor` in the advanced settings tab of your Database Instance dashboard in the Scaleway console. You can replace `auditor` with the name of your principal audit role.
```
Comment (SamyOubouaziz) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:285]:
```suggestion
1. Set `pgaudit.log` to `FUNCTION,ROLE,DDL,MISC,MISC_SET`, and the `pgaudit.role` as `auditor` in the advanced settings tab of your Database Instance dashboard in the Scaleway console. You can replace `auditor` with the name of your principal audit role.
```

File: managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx [hunk 27]
```diff
@@ -0,0 +1,303 @@
...
+      to auditor;
+    ```
+    Replace `mydb` with the name of your database and `mytable` with the name of the table.
+
+
+
+
+
+
+
```
Comment (ldecarvalho-doc) [inline managed-databases/postgresql-and-mysql/api-cli/pg-audit.mdx:303]:
```suggestion
```

**Review by SamyOubouaziz — APPROVED** [review]

**Pull request merged**